src/OrchardCore/OrchardCore.Data.YesSql/DbConnectionValidator.cs

```diff
@@ -72,32 +73,72 @@ public async Task<DbConnectionValidatorResult> ValidateAsync(string databaseProv
         }
 
         var selectBuilder = GetSelectBuilderForDocumentTable(tablePrefix, providerName);
-
         try
         {
             var selectCommand = connection.CreateCommand();
             selectCommand.CommandText = selectBuilder.ToSqlString();
 
             using var result = await selectCommand.ExecuteReaderAsync();
-
-            // at this point the query succeeded and the table exists
-            return DbConnectionValidatorResult.DocumentFound;
+            if (!isDefaultShell)
+            {
+                // The 'Document' table exists.
+                return DbConnectionValidatorResult.DocumentTableFound;
+            }
+
+            var columns = Enumerable.Range(0, result.FieldCount).Select(result.GetName);
+            if (!columns.Any(c => c == "Type") || !columns.Any(c => c == "Content") || !columns.Any(c => c == "Version"))
```

Comment:
Place all the or statements inside the same Any predict instead of having to check the columns all over again every time

Reply:
Good point

Reply:
I think the logic would be columns.Any(c => c != "Type" && c != …)

Reply:
Done but reverted as it is wrong

Reply:
@jtkech I noticed it. That is why I posted my previous comment. Please review my last statement. I have one more suggestion I'll level a separate comment in 1 mins

Reply:
Yes we could check that any column are equal to one of the names list, but then we would also need to check the `Id`, but I opted to not rely on the creation of a new column, e.g. the Version column was not existing at the beginning.

Otherwise we could check the fieldCount value and check that all columns are there and at the right place.

Reply:
@jtkech I think the statement `if (columns.Any(c => c != "Type" && c != "Content" && c != "Version"))** does the same thing as what you are doing.

Reply:
Not really as we would need to also do `c != "Id"`, and then add others if a new column is added to the Document table, at the beginning I wanted to just check the `Type` column because if then we found a Type value equal to the ShellDescriptor type we could assume that it is the ShellDescriptor document.

But here I have another idea to onlly do one lookup.

Reply:
You could use intersect and check if the length = 3.

Reply:
Done but not tried yet

Reply:
Yes I did something like this

Reply:
What you did looks good to me. I would think we should also add "Id" column and make it != 4 but it's up to you.

Either way, the code looks good to me. I have not tested it. But, if you tested it, I say you merge it...

Reply:
Okay I will also check `Id` and retry it.

Reply:
That looks much better